### PR TITLE
`set_tempdir` utils, abstract multi_ep tests

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
- Introduce a new `set_tempdir` util that sets the `tempfile.tempdir` if provided. This helps us remove duplicated code from the repository.
- Add missing package data for python environment system.
- Abstract the engine, workflow creation and run in the multi-ep test. This logic is same for all systems so there is a lot of duplicate code.
- Clean up aml system unit test

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/common/utils.py [hunk 1]
```diff
@@ -327,3 +328,18 @@ def copy_dir(src_dir, dst_dir, ignore=None, **kwargs):
                 f"Received shutil.Error '{e}' but all required file names exist in destination directory. "
                 "Assuming all files were copied successfully and continuing."
             )
+
+
+def set_tempdir(tempdir: str = None):
+    """Set the root directory for tempfiles.
+
+    :param new_tempdir: new tempdir.
```
Comment (trajepl) [inline olive/common/utils.py:336]: Align the param name in comment with the `tempdir`.

Comment (jambayk) [reply olive/common/utils.py]: Oh I missed this. Thanks for the catch!
Updated

File: olive/common/utils.py [hunk 2]
```diff
@@ -327,3 +328,18 @@ def copy_dir(src_dir, dst_dir, ignore=None, **kwargs):
...
                 "Assuming all files were copied successfully and continuing."
             )
+
+
+def set_tempdir(tempdir: str = None):
+    """Set the root directory for tempfiles.
+
+    :param tempdir: new tempdir.
+    """
+    if tempdir is None:
```
Comment (shaahji) [inline olive/common/utils.py:338]: nit: May be check for empty as well.

File: test/multiple_ep/utils.py [hunk 3]
```diff
@@ -65,3 +69,29 @@ def download_data():
...
+
+
+def create_and_run_workflow(tmp_path, system_config, execution_providers, model_config, metric, only_target=False):
+    # use the olive managed python environment as the test environment
+    output_dir = tmp_path / "output"
+    output_dir.mkdir()
+
+    evaluator_config = OliveEvaluatorConfig(metrics=[metric])
+
+    cache = tmp_path / "cache"
```
Comment (shaahji) [inline test/multiple_ep/utils.py:81]: nit: To be consistent, rename it to 'cache_dir'

**Review by shaahji — APPROVED** [review]

**Pull request merged**